src/lang_c/libowi/src/owi.c

```diff
@@ -16,12 +16,27 @@
     _Static_assert(sizeof(T) == 1 || sizeof(T) == 2 || sizeof(T) == 4 ||       \
                        sizeof(T) == 8,                                         \
                    "Unsupported type size!");                                  \
-    (sizeof(T) == 1)   ? (T)(owi_uint8())                                      \
-    : (sizeof(T) == 2) ? (T)(owi_uint16())                                     \
+    (sizeof(T) == 1)   ? (T)(owi_uint32())                                     \
+    : (sizeof(T) == 2) ? (T)(owi_uint32())                                     \
     : (sizeof(T) == 4) ? (T)(owi_uint32())                                     \
                        : (T)(owi_uint64());                                    \
```

Comment:
why not reduce this to: `sizeof(T) <= 4 ?  (T)(owi_uint32()) : (T)(owi_uint64())` ?